src/Domain/Tags/Faces.hpp

```diff
@@ -0,0 +1,119 @@
+};
+
+/// Wrap all tags in `TagsList` in `domain::Tags::Faces`, except those in the
+/// `VolumeTags` list
+template <size_t Dim, typename TagsList, typename VolumeTags = tmpl::list<>>
+using make_faces_tags =
+    tmpl::transform<TagsList, make_faces_tag<tmpl::pin<tmpl::size_t<Dim>>,
+                                             tmpl::_1, tmpl::pin<VolumeTags>>>;
+
+}  // namespace domain
+
+namespace db {
+namespace detail {
+template <size_t Dim, typename VariablesTag>
+struct FacesSubitemsImpl {
+  template <typename LocalTag>
+  using faces_tag = domain::Tags::Faces<Dim, LocalTag>;
+
+  using tag = faces_tag<VariablesTag>;
+  using type =
+      db::wrap_tags_in<faces_tag, typename VariablesTag::type::tags_list>;
+
+  template <typename Subtag>
+  static void create_item(
+      const gsl::not_null<typename tag::type*> parent_value,
+      const gsl::not_null<typename Subtag::type*> sub_value) noexcept {
+    sub_value->clear();
+    for (auto& [direction, all_parent_vars] : *parent_value) {
+      auto& parent_var = get<typename Subtag::tag>(all_parent_vars);
+      auto& sub_var = (*sub_value)[direction];
+      for (auto vars_it = parent_var.begin(), sub_var_it = sub_var.begin();
+           vars_it != parent_var.end(); ++vars_it, ++sub_var_it) {
+        sub_var_it->set_data_ref(&*vars_it);
+      }
+    }
+  }
+
+  // The `return_type` can be anything for Subitems because the DataBox figures
+  // out the correct return type, we just use the `return_type` type alias to
+  // signal to the DataBox we want mutating behavior.
+  using return_type = NoSuchType;
+
+  template <typename Subtag>
+  static void create_compute_item(
+      const gsl::not_null<typename Subtag::type*> sub_value,
+      const typename tag::type& parent_value) noexcept {
+    for (const auto& [direction, all_parent_vars] : parent_value) {
+      const auto& parent_var = get<typename Subtag::tag>(all_parent_vars);
+      auto& sub_var = (*sub_value)[direction];
+      auto sub_var_it = sub_var.begin();
+      for (auto vars_it = parent_var.begin(); vars_it != parent_var.end();
+           ++vars_it, ++sub_var_it) {
+        // clang-tidy: do not use const_cast
+        // The DataBox will only give out a const reference to the
+        // result of a compute item.  Here, that is a reference to a
+        // const map to Tensors of DataVectors.  There is no (publicly
+        // visible) indirection there, so having the map const will
+        // allow only allow const access to the contained DataVectors,
+        // so no modification through the pointer cast here is
+        // possible.
+        sub_var_it->set_data_ref(const_cast<DataVector*>(&*vars_it));  // NOLINT
+      }
+    }
+  }
+};
+}  // namespace detail
+
+template <size_t Dim, typename VariablesTag>
+struct Subitems<domain::Tags::Faces<Dim, VariablesTag>,
+                Requires<tt::is_a_v<Variables, typename VariablesTag::type>>>
+    : detail::FacesSubitemsImpl<Dim, VariablesTag> {};
```

Comment:
Is there a reason for the separate impl struct?  Could you just put the body here?